R/utils.R

```diff
@@ -28,4 +28,34 @@ check_response <- function(link){
   
   
   textConnection(rawToChar(response$content))
+}
+
+
+
+with_cache <- function(cache, file, memoised, unmemoised) {
+  # cache in memory
+  if (cache && is.null(file)) {
+    return(memoised())
+  }
+  
+  # cache in file  
+  if (cache && file.exists(file)) {
+    return(suppressMessages(readr::read_csv(file)))
```

Comment:
How attached are you to the `readr` dependency? I love tibbles but kept the package some generalized (to accomodate those that don't) with this line: https://github.com/boshek/rsoi/blob/cdfc6da5013dfb3f30e2062fe2180bb70cd49081/R/download-oni.R#L67

Reply:
Not at all. The only reason I use readr there is because read_csv parses everything automatically so that the original dataframe is recovered identically. read.csv, ONTH, messes up the formating and adds a rownumber column. But, no problem. I'll tweak things to get the same behaviour with base R to keep the dependencies lean.